metacat-common-server/src/main/java/com/netflix/metacat/common/server/partition/util/PartitionUtil.java

```diff
@@ -57,9 +58,10 @@ public static void getPartitionKeyValues(final String location, final Map<String
                 // value is null/empty string or any other values that cannot be escaped.
                 //
                 if (values[0].equalsIgnoreCase("null")
-                    || values[1].equalsIgnoreCase("null")
-                    || values[1].equalsIgnoreCase(DEFAULT_PARTITION_NAME)) {
+                    || values[1].equalsIgnoreCase("null")) {
                     log.debug("Found 'null' string in kvp [{}] skipping.", part);
+                } else if (values[1].equalsIgnoreCase(DEFAULT_PARTITION_NAME)) {
```

Comment:
I don't quite understand the logic. Are we saying that if null we will skip but if DEFAULT_PARTITION_NAME we will return null? I guess DEFAULT_PARTITION_NAME is the internal representation of null in hive?

Reply:
Yes, __HIVE_DEFAULT_PARTITION__ is the internal representation of null in hive.